lms/djangoapps/grades/models.py

```diff
@@ -0,0 +1,269 @@
+"""
+Models used for robust grading.
+
+Robust grading allows student scores to be saved per-subsection independent
+of any changes that may occur to the course after the score is achieved.
+"""
+
+from base64 import b64encode
+from collections import namedtuple
+from hashlib import sha256
+import json
+import logging
+from operator import attrgetter
+
+from django.db import models
+from django.db.utils import IntegrityError
+from model_utils.models import TimeStampedModel
+
+from coursewarehistoryextended.fields import UnsignedBigIntAutoField
+from xmodule_django.models import CourseKeyField, UsageKeyField
+
+
+log = logging.getLogger(__name__)
+
+
+# Used to serialize information about a block at the time it was used in
+# grade calculation.
+BlockRecord = namedtuple('BlockRecord', ['locator', 'weight', 'max_score'])
+
+
+class BlockRecordList(list):
+    """
+    An ordered collection of BlockRecord objects.
+    """
+    def to_json(self):
+        """
+        Return a JSON-serialized version of the list of block records, using a
+        stable ordering.
+        """
+        sorted_blocks = sorted(self, key=attrgetter('locator'))
+        return json.dumps([block._asdict() for block in sorted_blocks], separators=(',', ':'), sort_keys=True)
```

Comment:
Thanks for the clarification on the `separators`.  Can you add a comment that we're explicitly choosing to discard space as a delimiter?  It's otherwise difficult to read that space was excluded.